lib/fotofetch.rb

```diff
@@ -59,25 +81,27 @@ def height_ok?(sizes)
       (0 < height && height < sizes[1][1]) || height > (sizes[1][1] * 2)
     end
 
+    def root_url(link)
+      link.split("/")[2]
+    end
+
     # Adds root urls as hash keys
     def add_sources(urls)
-      results = {}.compare_by_identity
-      urls.each { |link| results[link.split("/")[2]] = link}
-      results
+      urls.map { |link| [root_url(link), link] }.to_h
```

Comment:
@SteveOscar sorry about that - should be good now

I definitely did that unintentionally.. However, I will say that it seems like a strange use of a hash. If a hash key needs to be repeated, seems like it's time to reach for something else. Maybe an array of hashes like:
[{ root_url: <root1>, link: <link1> }, { root_url: <root2>, link: <link2>}, { root_url: <root2>, link: <link3>}...]

Anyways thanks for opening this up for contributions.